lib/cli/commands/restore.ts

```diff
@@ -28,7 +44,47 @@ const formatOutput = (response: RestoreNodeResponse.AsObject) => {
   console.log(walletRestoredMessage);
 };
 
-export const handler = (argv: Arguments<any>) => {
+export const handler = async (argv: Arguments<any>) => {
+  const request = new RestoreNodeRequest();
+
+  if (argv.backup_directory && argv.backup_directory !== '') {
+    const readPromises: Promise<void>[] = [];
```

Comment:
It might make sense to load the backup files only after the mnemonic and password are entered and validated. My concern is reading the backup might take a little while when the user should be prompted for the mnemonic right away.

Another approach would be to define the `readPromises` outside of this scope. Make the `readFile` calls here as you're already doing, but then don't `await Promise.all(readPromises)` until after the mnemonic is entered and validated. The downside is that in case of a read error the process would exit after the mnemonic prompt is displayed.